File: scm-csharp/scheme/primitives/PrimitiveRandomAccessFileSize.cs
namespace scheme;

public class PrimitiveRandomAccessFileSize : Primitive
{
    public override string Name() => "random-access-file-size";
    public override string Info() =>
        "Syntax: (random-access-file-size f)\n" +
        "Library: (scm random access)\n" +
        "Description: Returns the current size of random-access file f in bytes.\n" +
        "Example:\n" +
        "  (random-access-file-size f) => 1024";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        var f = RandomAccessFileHandle.Of(pos, arguments[0], Name());
        try
        {
            return f.Size();
        }
        catch (SchemeError) { throw; }
        catch (System.Exception e)
        {
            throw new SchemeError(pos, Name() + ": io failure: ~a", e.Message);
        }
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveRandomAccessFileSize.java
package scheme.primitives;

import scheme.*;

public class PrimitiveRandomAccessFileSize extends Primitive {
    @Override public String name() { return "random-access-file-size"; }
    @Override public String info() {
        return "Syntax: (random-access-file-size f)\n" +
               "Library: (scm random access)\n" +
               "Description: Returns the current size of random-access file f in bytes.\n" +
               "Example:\n" +
               "  (random-access-file-size f) => 1024";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        RandomAccessFileHandle f = RandomAccessFileHandle.of(pos, arguments[0], name());
        try {
            return f.size();
        } catch (SchemeError e) {
            throw e;
        } catch (Exception e) {
            throw new SchemeError(pos, name() + ": io failure: ~a", e.getMessage());
        }
    }
}
